app/uk/gov/hmrc/cataloguefrontend/connector/TeamsAndRepositoriesConnector.scala

```diff
@@ -209,37 +214,50 @@ class TeamsAndRepositoriesConnector @Inject()(
     http.GET[Seq[Team]](teamsAndServicesBaseUrl + s"/api/teams")
 
   def allDigitalServices(implicit hc: HeaderCarrier): Future[Seq[String]] =
-    http.GET[Seq[String]](teamsAndServicesBaseUrl + s"/api/digital-services")
+    http.GET[Seq[String]](teamsAndServicesBaseUrl + s"/api/digital-services" +
+      queryString(teamsAndRepositoriesArchived.toList))
 
   def teamInfo(teamName: TeamName)(implicit hc: HeaderCarrier): Future[Option[Team]] =
     http
-      .GET[Option[Team]](teamsAndServicesBaseUrl + s"/api/teams_with_details/${encodePathParam(teamName.asString)}")
+      .GET[Option[Team]](teamsAndServicesBaseUrl + s"/api/teams_with_details/${encodePathParam(teamName.asString)}" +
+        queryString(teamsAndRepositoriesArchived.toList))
       .recover {
         case _ => None
       }
 
   def teamsWithRepositories(implicit hc: HeaderCarrier): Future[Seq[Team]] =
-    http.GET[Seq[Team]](teamsAndServicesBaseUrl + s"/api/teams_with_repositories")
+    http.GET[Seq[Team]](teamsAndServicesBaseUrl + s"/api/teams_with_repositories" +
+      queryString(teamsAndRepositoriesArchived.toList))
 
   def digitalServiceInfo(digitalServiceName: String)(implicit hc: HeaderCarrier): Future[Option[DigitalService]] = {
     val url = teamsAndServicesBaseUrl + s"/api/digital-services/${encodePathParam(digitalServiceName)}"
     http.GET[Option[DigitalService]](url)
   }
 
   def allRepositories(implicit hc: HeaderCarrier): Future[Seq[RepositoryDisplayDetails]] =
-    http.GET[Seq[RepositoryDisplayDetails]](teamsAndServicesBaseUrl + s"/api/repositories")
+    http.GET[Seq[RepositoryDisplayDetails]](teamsAndServicesBaseUrl + s"/api/repositories" +
+      queryString(teamsAndRepositoriesArchived.toList))
 
   def repositoryDetails(name: String)(implicit hc: HeaderCarrier): Future[Option[RepositoryDetails]] =
     http.GET[Option[RepositoryDetails]](teamsAndServicesBaseUrl + s"/api/repositories/${encodePathParam(name)}")
 
   def teamsByService(serviceNames: Seq[String])(implicit hc: HeaderCarrier): Future[Map[ServiceName, Seq[TeamName]]] =
     http.POST[JsValue, Map[ServiceName, Seq[TeamName]]](
-      teamsAndServicesBaseUrl + s"/api/services?teamDetails=true",
+      teamsAndServicesBaseUrl + s"/api/services" +
+        queryString(teamsAndRepositoriesArchived.toList :+ ("teamDetails", "true")),
       Json.arr(serviceNames.map(toJsFieldJsValueWrapper(_)): _*)
     )
 
   def allTeamsByService()(implicit hc: HeaderCarrier): Future[Map[ServiceName, Seq[TeamName]]] =
-    http.GET[Map[ServiceName, Seq[TeamName]]](teamsAndServicesBaseUrl + s"/api/services?teamDetails=true")
+    http.GET[Map[ServiceName, Seq[TeamName]]](teamsAndServicesBaseUrl + s"/api/services" +
+      queryString(teamsAndRepositoriesArchived.toList :+ ("teamDetails", "true")))
+
+  private def queryString(parameters: Seq[(String, String)]): String = {
+    parameters match {
+      case Nil => ""
+      case _   => "?" + parameters.map(param => s"${param._1}=${param._2}").mkString("&")
```

Comment:
`.mkString("?", "&", "")`

Reply:
However the http-verbs functions take a separate param for query params with type "Map[String, Seq[String]]" - which will do this for you. This is preferable to building into the url.

Reply:
@colin-lamed Holy hell, I never knew that alternative version of "mkString" - that is SO USEFUL! However, will update using http-verbs.